### DISCUSSION TITLE
Templated Controls not visible in the Designer

### DISCUSSION THREAD

**Discussion by DefaultLP** [AvaloniaUI/Avalonia#11064] [Q&A] [answered]
**Describe the bug**
My Templated Controls are only visible in the Build but on in the Designer. 
They do not take up any space unless `Width` or `Height` is explicitly specified:
`<controls:CategoryControl/>`
Takes up no space in Designer.
`<controls:CategoryControl Height="20"/>`
Does take up space. But still not visible. (No setting a Background color does not work either)
When the control is in a DockPanel it does however take up space [img 1].

They are visible however in their own Designer [img 2].

**To Reproduce**
1. Create Templated Control in a folder called "Controls"
2. Include the Style in `Application.Styles`
3. Add the controls namespace to any window of other control
4. Build and run the app

**Expected behaviour**
The Templated Controls should be visible in the Designer.

**Screenshots**
Left is the Designer and right is the build app
<img width="345" alt="Asset_Manager_9hzfEdeUgq" src="https://user-images.githubusercontent.com/29122114/233064130-541cb16f-6b7b-4376-b837-b25b31a02f48.png">
The templated control is visible in its own designer
<img width="1630" alt="rider64_yOtlMzsPLf" src="https://user-images.githubusercontent.com/29122114/233064163-4d7296b2-481a-428f-af3f-9698ebb62663.png">

**Desktop (please complete the following information):**

- OS: Windows 11 Pro
- Version: 22H2
- OS Build: 22621.1555

**Additional context**
There were loads of ideas I already tried.
Q: **Include it in another view or window.**
A: Did not show either

Q: I**nclude the style directly in the window instead of App.axaml**
A: Gives me an URL parsing error.
```
<Window.Styles>
    <StyleInclude Source="/../Controls/Explorer/CategoryControl.axaml"/>
</Window.Styles>
```
Q: **Add Styles directly to the control.**
A: Did not show either. 

Q: **Switch to another IDE.**
A: I tried using Visual Studio instead of Rider but to no avail. Same problem

Q: **Update or revert Rider version**
A: I was using Rider 2022.3 before and updated it to 2023.1. Same issue with both of them.

Q: **Have you tried touching grass?**
A: It is currently raining outside, so no.

**Comment by timunie** [comment]
@DefaultLP I assume IconFill and IconData are set and not only loaded at runtime?

**Comment by DefaultLP** [comment]
@timunie Correct. They are more a convenience for me when designing. I removed both of them to test if they were causing the problem. But sadly still nothing.

I also tried it with a Templated Control that had no bindings, to begin with. Just a border and some text. But still nothing.
<img width="322" alt="Asset_Manager_dbBsObO0aJ" src="https://user-images.githubusercontent.com/29122114/233072001-a38723bd-8ded-4b4a-b628-a0ee567dd187.png">
<img width="1411" alt="rider64_iELdSWLqks" src="https://user-images.githubusercontent.com/29122114/233071982-353ae770-94f1-4e7a-932f-60049172587b.png">

**Comment by timunie** [comment]
Can you file a minimal sample? Just like shown above with nothing more?

**Comment by DefaultLP** [comment]
Sure. Give me a moment

**Comment by DefaultLP** [comment, marked as answer]
After a lot of trial and error, I was **not** able to reproduce the bug. I even went as far as to literally copy most of my code over and it still works. 
<img width="379" alt="TestApp_qYgBhT4juN" src="https://user-images.githubusercontent.com/29122114/233078992-7faec40a-8ebf-49f4-9969-6426408da64c.png">

I also looked at the .csproj files of both projects. They are completely identical.
(And also the .snl files but I doubt they had anything to do with this in the first place)

Something in this first project was probably messed up by me trying out things with Avalonia.

**Comment by DefaultLP** [comment]
I quickly copied the original project onto my laptop and the Template Controls were still not showing. It is definitely a project-related issue but I cannot find the difference between the old and new projects which cause this bug. 

Deleting both the "bin" and obj" folders and rebuilding also did nothing. 
While I can confidently say that Avalonia is not at fault here, I cannot for the life of me find or reproduce this bug.

Thank you @timunie for taking the time and helping me with this.

**Comment by timunie** [comment]
Sometimes also some old cached files can be an issue. deleting `bin, obj and .vs` folder can help. Anyway, I'll convert this issue into a Q&A now as it's not an Avalonia bug.
